edx_lint/pylint/annotations_check.py

```diff
@@ -362,15 +369,41 @@ class FeatureToggleAnnotationChecker(AnnotationBaseChecker):
             NON_BOOLEAN_DEFAULT_VALUE,
             "Feature toggle default values must be boolean",
         ),
+        ("E%d64" % BASE_ID): (
+            "missing feature toggle annotation",
+            MISSING_ANNOTATION,
+            (
+                "When a WaffleFlag/Switch object is created, a corresponding annotation must be present above in the"
+                " same module and with a matching name",
+            )
+        ),
     }
 
+    TOGGLE_FUNC_NAMES = [
+        "WaffleFlag",
```

Comment:
Oops. @regisb: I just noticed this list is missing all `LegacyWaffleSwitch` and `LegacyWaffleFlag`. I don't think we should wait for the legacy classes to be retired before asking for annotations.

Reply:
Right. This will make the linter slightly more complicated (you'll see why). But I think I can add it to this PR.